### PR TITLE
August 28 Update - Comment Incentives

### PR DESCRIPTION
**Pull request by pavlovcik**
This one has full comment incentive support and must be merged in by [28th August (Monday)](https://t.me/c/1588400061/3250/4373)

### CODE REVIEW

File: ubiquibot-config-default.json [hunk 1]
```diff
@@ -93,37 +93,12 @@
...
-    "footnoteReference": 0,
-    "table": 0,
-    "tableCell": 0,
-    "tableRow": 0,
-    "alignType": 0,
-    "flowContent": 0,
-    "listContent": 0,
-    "phrasingContent": 0,
-    "rowContent": 0,
-    "tableContent": 0
```
Comment (pavlovcik) [inline ubiquibot-config-default.json:127]: These should all be initialized to `0` why was this removed?

Comment (whilefoo) [reply ubiquibot-config-default.json:127]: Because they every element is 0 by default

File: src/handlers/payout/post.ts [hunk 2]
```diff
@@ -1,18 +1,21 @@
 import { getWalletAddress } from "../../adapters/supabase";
 import { getBotConfig, getBotContext, getLogger } from "../../bindings";
 import { addCommentToIssue, generatePermit2Signature, getAllIssueComments, getIssueDescription, getTokenSymbol, parseComments } from "../../helpers";
-import { MarkdownItem, Payload, UserType, CommentElementPricing } from "../../types";
+import { Incentives, Payload, StateReason, UserType } from "../../types";
+import { commentParser } from "../comment";
+import Decimal from "decimal.js";
+import { bountyInfo } from "../wildcard";
 
-const ItemsToExclude: string[] = [MarkdownItem.BlockQuote];
```
[the remark this answers is not in the document]
Comment (whilefoo) [reply src/handlers/payout/post.ts:6]: we don't use Markdown anymore, so I didn't want it to be implying it's markdown

File: ubiquibot-config-default.json (hunk 1 above)
Comment (pavlovcik) [reply ubiquibot-config-default.json:127]: This is the `default` configuration so I thought it makes more sense to explicitly set defaults here instead of inferring them inside of the loader?

Comment (rndquu) [reply ubiquibot-config-default.json:127]: I agree that the default incentives config should be explicit because right now it is not clear what parameters are available

This [one](https://github.com/inikulin/parse5/blob/7f8c43c957ea60461d3afd55ca1905c095452f40/packages/parse5/lib/common/html.ts#L33) seems to be a comprehensive list of all available tags (supported by [parse5](https://parse5.js.org/index.html))

Adding all of the tags to the config is redundant but we can add at least most commonly used like `h1`, `img`, `label`, etc...

@whilefoo What do you think?

Comment (whilefoo) [reply ubiquibot-config-default.json:127]: Ok I agree, we can have the most common tags so it will be easier for partners to understand

File: .github/ubiquibot-config.yml [hunk 3]
```diff
@@ -33,6 +33,17 @@ default-labels:
...
+  comment:
+    elements:
+      code: 5
+      img: 5
+      h1: 1
+      li: 0.5
+      a: 0.5
+      blockquote: 0
+    totals:
+      word: 0.1
```
[the remark this answers is not in the document]
Comment (whilefoo) [reply .github/ubiquibot-config.yml:47]: Sure, shall I create a PR?

File: ubiquibot-config-default.json (hunk 1 above)
Comment (rndquu) [reply ubiquibot-config-default.json:127]: > Ok I agree, we can have the most common tags so it will be easier for partners to understand. Shall I create a PR?

This would be awesome, thank you

Comment (pavlovcik) [reply ubiquibot-config-default.json:127]: > I agree that the default incentives config should be explicit because right now it is not clear what parameters are available
> 
> 
> 
> This [one](https://github.com/inikulin/parse5/blob/7f8c43c957ea60461d3afd55ca1905c095452f40/packages/parse5/lib/common/html.ts#L33) seems to be a comprehensive list of all available tags (supported by [parse5](https://parse5.js.org/index.html))
> 
> 
> 
> Adding all of the tags to the config is redundant but we can add at least most commonly used like `h1`, `img`, `label`, etc...
> 
> 
> 
> @whilefoo What do you think?

Wait I already fixed this in my pull request. I need to find it which is tedious from my phone, but if you read this comment before I find it, please check the open pull requests from four days ago. Check my newest one that is not a draft.

File: src/handlers/payout/post.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xcodercrane) [reply src/handlers/payout/post.ts:6]: https://github.com/ubiquity/ubiquibot/commit/66ff9d36fa51b270dc11eb7a0c3146c1bf2cfce8

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]

File: src/handlers/payout/post.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (whilefoo) [reply src/handlers/payout/post.ts:6]: Actually we can use [tag enum](https://github.com/inikulin/parse5/blob/7f8c43c957ea60461d3afd55ca1905c095452f40/packages/parse5/lib/common/html.ts#L33) directly from parse5

[the remark this answers is not in the document]
Comment (pavlovcik) [reply src/handlers/payout/post.ts:6]: Most of those tags are not supported by GitHub though, I would prefer to do only what is compatible with GitHub comments.

**Review by 0xcodercrane — APPROVED** [review]
